pkg/server/service_unix.go

```diff
@@ -67,3 +70,51 @@ func (c *criService) initPlatform() error {
 func (c *criService) cniLoadOptions() []cni.CNIOpt {
 	return []cni.CNIOpt{cni.WithLoNetwork, cni.WithDefaultConf}
 }
+
+// runningInUserNS detects whether we are currently running in a user namespace.
+// Originally copied from github.com/lxc/lxd/shared/util.go
+func runningInUserNS() bool {
```

Comment:
containerd/sys.RunningInUserNS

Reply:
There was some discussion in an earlier PR about CRI not depending too much on containerd (see https://github.com/containerd/containerd/pull/4096#issuecomment-596778362), which is why I added it here; let me push a commit to see what it'd look like if we did that (so that we can discuss)

Reply:
archive and diff use that function